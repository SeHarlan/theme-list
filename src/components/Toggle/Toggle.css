.Toggle {
  input {
    display: none;
  }
  label {
    display: block;
    height: 1.5rem;
    width: 3.1rem;
    padding: 2px;
    border-radius: 1.5rem;
    border: 5px inset gray;
    background-color: rgb(65, 75, 80);
  }
  label::before {
    content: '';

    display: block;
    margin-top: -1px;
    
    height: 1.5rem;
    width: 1.5rem;

    background-color: grey;
    border: 1px gray outset;
    border-radius: 1.5rem;
    transition: 0.3s;
  }
  input:checked + label {
    background-color:  rgb(211, 217, 225);
    border: 5px inset gold;
    &::before {
      margin-left: 1.5rem;
      background-color: gold;
    }
  }
}