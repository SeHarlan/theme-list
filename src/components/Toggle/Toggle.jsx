import React from 'react';
import PropTypes from 'prop-types';
import style from './Toggle.css';

const Toggle = ({ action, checked }) => {
  return (
    <section className={style.Toggle} >
      <input type="checkbox" id="toggle" name="toggle" onChange={action} checked={checked} />
      <label htmlFor="toggle"></label>
    </section>
  );};

Toggle.propTypes = {
  checked: PropTypes.bool.isRequired,
  theme: PropTypes.string.isRequired,
  action: PropTypes.func.isRequired
};

export default Toggle;
