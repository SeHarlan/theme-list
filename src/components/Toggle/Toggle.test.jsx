import React from 'react';
import { shallow } from 'enzyme';
import Toggle from './Toggle';

describe('Toggle component', () => {
  it('renders Toggle', () => {
    const wrapper = shallow(<Toggle const checked={true} theme="Alive" />);
    expect(wrapper).toMatchSnapshot();
  });
});
