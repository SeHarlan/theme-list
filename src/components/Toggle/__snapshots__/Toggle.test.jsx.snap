// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle component renders Toggle 1`] = `
<section
  className="Toggle"
>
  <input
    checked={true}
    id="toggle"
    name="toggle"
    type="checkbox"
  />
  <label
    htmlFor="toggle"
  />
</section>
`;
